src/main/java/wolox/training/models/User.java

```diff
@@ -0,0 +1,100 @@
+package wolox.training.models;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import wolox.training.exceptions.users.BookAlreadyOwnedException;
+
+@Entity
+@Table(name = "users")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+
+    @Column(nullable = false)
+    private String username;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Column(nullable = false)
+    private LocalDate birthday;
+
+    @ManyToMany(cascade = CascadeType.ALL)
+    @JoinTable(name = "book_user",
+            joinColumns = @JoinColumn(name = "book_id", referencedColumnName = "id"),
+            inverseJoinColumns = @JoinColumn(name = "user_id", referencedColumnName = "id"))
+    private List<Book> books;
+
+    public User() {
+        this.books = Collections.emptyList();
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public void setId(long id) {
+        this.id = id;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public LocalDate getBirthday() {
+        return birthday;
+    }
+
+    public void setBirthday(LocalDate birthday) {
+        this.birthday = birthday;
+    }
+
+    public List<Book> getBooks() {
+        return this.books;
+    }
+
+    public void setBooks(List<Book> books) {
+        this.books = books;
+    }
+
+    public void addBook(Book newBook) {
+        checkIfAlreadyExists(newBook);
+        this.books.add(newBook);
+    }
+
+    public void addBooks(List<Book> books) {
+        books.forEach(this::checkIfAlreadyExists);
+        this.books.addAll(books);
+    }
+
+    private void checkIfAlreadyExists(Book newBook) {
+        Book bookFound = this.books.stream().filter(book -> book.getId() == newBook.getId()).findAny().orElse(null);
+        if (bookFound != null) {
+            throw new BookAlreadyOwnedException("Book with id "+newBook.getId()+" already exists on User");
```

Comment:
instead of using .orElse(null) and then checking if it's not null, you could do this: 

```
Book bookFound = this.books.stream()
        .filter(book -> book.getId() == newBook.getId())
        .findAny()
        .ifPresent(throw new ...)
```

I recommend adding each method of the stream (filter, findAny, ifPresent, etc.) in a new line to make it easier to read 😄

Reply:
More cleaned. Thx!